### PR TITLE
Better Janitor Error Handling // Index Integrity Verification

### PR DESCRIPTION
**Pull request by Miserlou**
## Issue Number
n/a

## Purpose/Implementation Notes
Small fixes, one to make the Janitor slightly more robust but slightly less aggressive, and one more to ensure that indexes aren't corrupt after download.

## Types of changes
- Bugfix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: foreman/data_refinery_foreman/foreman/main.py [hunk 1]
```diff
@@ -447,6 +447,16 @@ def retry_hung_processor_jobs() -> None:
...
+        except TypeError:
+            # Almost certainly a python-nomad issue:
+            # File "/usr/local/lib/python3.5/dist-packages/nomad/api/job.py", line 63, in get_job
+            #   return self.request(id, method="get").json()
+            # File "/usr/local/lib/python3.5/dist-packages/nomad/api/base.py", line 74, in request
+            #   endpoint = self._endpoint_builder(self.ENDPOINT, *args)
+            # File "/usr/local/lib/python3.5/dist-packages/nomad/api/base.py", line 28, in _endpoint_builder
+            #   u = "/".join(args)
+            # TypeError: sequence item 1: expected str instance, NoneType found
+            logger.info("Couldn't query Nomad about Processor Job.", processor_job=job.id)
```
Comment (kurtwheeler) [inline foreman/data_refinery_foreman/foreman/main.py]: I've seen this before. I think it happens when you pass None to get_job()

File: workers/data_refinery_workers/processors/janitor.py [hunk 2]
```diff
@@ -59,21 +59,32 @@ def _find_and_remove_expired_jobs(job_context):
...
+            except ProcessorJob.DoesNotExist:
+                # This job has vanished from the DB - clean it up!
+                logger.exception("Janitor found no record of " + item + " - why?")
+                pass
             except Exception:
                 # If we don't have a record of the job, we don't need the directory.
-                pass
-
+                logger.exception("Problem finding job record for " + item + " - why?")
+                continue
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/janitor.py]: The comment above this now seems out of date.

Comment (Miserlou) [reply workers/data_refinery_workers/processors/janitor.py]: There are two separate cases here, that we are handling in different ways. The first (object DNE) - then we say we found no record of it, and we `pass` so it gets deleted. If there is another type of Exception, that means that we are likely currently unable to access the database (PgBouncer is down, for example), so we don't want to perform any destructive action yet, hence the continue and "problem finding record".

Comment (kurtwheeler) [reply workers/data_refinery_workers/processors/janitor.py]: Right, but I'm talking about the `# If we don't have a record of the job, we don't need the directory.` comment which used to precede the `pass`, but now it's before a continue so I don't think it makes sense.

**Pull request merged**